tempesta_fw/cfg.c

```diff
@@ -888,6 +888,73 @@ tfw_cfg_map_enum(const TfwCfgEnum mappings[],
 EXPORT_SYMBOL(tfw_cfg_map_enum);
 
 /**
+ * Get value of attribute with name @attr_key.
+ * Return @default_val if the attribute is not found in the entry @e.
+ */
+const char *
+tfw_cfg_get_attr(const TfwCfgEntry *e, const char *attr_key,
+		 const char *default_val)
+{
+	const char *key, *val;
+	size_t i;
+
+	TFW_CFG_ENTRY_FOR_EACH_ATTR(e, i, key, val) {
+		if (!strcasecmp(key, attr_key))
+			return val;
+	}
+
+	return default_val;
+}
+EXPORT_SYMBOL(tfw_cfg_get_attr);
+
+/**
+ * Check that integer is in specified range.
+ * Print an error and return non-zero if it is out of range.
+ */
+int
+tfw_cfg_check_range(long value, long min, long max)
+{
+	if (min != max && (value < min || value > max)) {
+		TFW_ERR("the value %ld is out of range: [%ld, %ld]\n",
```

Comment:
Well, I jsut wanted to put all error messages of the configuration parser to the single place, where it is easy to inspect and edit them. 

That is done because these error messages will be seen by end user. I mean, they are not some debugging messages for developers only, I think they will be pretty common messages seen by system administrators who edit configuration files. So likely we would like to work carefully on them in future. Also, by placing error messages here, we have aceess to parser internals and thus print additional verbose info: file/line/position/etc.

In addition, these functions reduce duplicate logic a little bit: this particular function `tfw_cfg_check_range()` is used in `tfw_be_cfg_add_srv()` and in `tfw_cfg_set_int()`. For me it is better to avoid such duplicate error messages because they are subject for change as I described above.

Hope you got my motivation.